django/applications/catmaid/middleware.py

```diff
@@ -104,7 +107,24 @@ def process_exception(self, request, exception):
             (exc_type, exc_info, tb) = sys.exc_info()
             response['info'] = str(exc_info)
             response['traceback'] = ''.join(traceback.format_tb(tb))
-        return JsonResponse(response)
+
+        # Some CATMAID errors have a more detailed status code
+        if isinstance(exception, ClientError):
+            status = exception.status_code
+        elif isinstance(exception, Http404):
+            status = 404
+        elif isinstance(exception, APIException):
+            status = exception.status_code
+        elif isinstance(exception, ValueError):
+            # Value errors are assumed to be problems with the request/client.
+            status = 400
+        else:
+            status = 500
+
+        # Allow easier access to the status code for clients
+        response['status_code'] = status
```

Comment:
Is this necessary? Anyone interested in the status code will have access to it from the raw response object. It doesn't harm to throw it in, of course, I just think it's blurring the line between the transport layer and the data received.

Reply:
Good point, thanks. After looking at it a second time I agree that it mixes different concepts. The use of the status code that motivated storing it in the first place should just not check for a status code, but for a defined error type (`settings-manager.js` checks for a 404). If it did, we wouldn't need to store in the first place and could use the status code only as optional error parsing informaiton when creating error types.  I'll fix this.

Reply:
I force-pushed a fix for this. The status code is now only used as an extra option when constructing error types.